data/src/main/java/com/example/data/implementationRepo/LastChosenCitiesRepoImpl.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.data.implementationRepo
+
+import com.example.data.db.Database
+import com.example.data.db.dao.CityDao
+import com.example.data.mappers.LastChosenCitiesEntityMapper
+import com.example.data.utils.MAX_TABLE_SIZE
+import com.example.domain.models.CityModel
+import com.example.domain.repositories.LastChosenCitiesRepo
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+
+class LastChosenCitiesRepoImpl(
+    private val cityDao: CityDao,
+    private val mapper: LastChosenCitiesEntityMapper,
+    private val database: Database
+) : LastChosenCitiesRepo {
+
+    override fun getLastChosenCities(): Single<List<CityModel>> {//This method gets list of last ten chosen cities from db and maps the list to list of CityModels
+        return cityDao.getAllLastChosenCities()
+            .subscribeOn(Schedulers.io())
+            .map { mapper.fromEntityToCityModelList(it) }
+    }
+
+    override fun insertCityToLastChosenCitiesEntity(
+        cityId: Int,
+        cityList: List<CityModel>?
+    ): Completable {
+        return Completable.fromCallable {
+            database.runInTransaction {
+                val tempCityDao = database.cityDao()
```

Comment:
Why tempCityDao? You already have cityDao here, from constructor.

Reply:
Fixed